web/src/message_notifications.ts

```diff
@@ -210,7 +210,7 @@ export function process_notification(notification: {
             notification_object.addEventListener("click", () => {
                 notification_object.close();
                 if (message.type !== "test-notification") {
-                    message_view.narrow_by_topic(message.id, {trigger: "notification"});
+                    window.location.href = hash_util.by_conversation_and_time_url(message);
```

Comment:
Do we intend to keep logging the `trigger` in this case ? If it's useful we have a similar case in [zoom_to_message_near](https://github.com/zulip/zulip/blob/9fe24279d7208233d003d1313abc95f654decf4d/web/src/hotkey.ts#L1473) - we can abstract out that logic as a function and use it at both the places.

Reply:
We do use `trigger` for some things, so it would be better to have it available. Abstracting that function sounds good.